streamer/pulp/streamer/server.py

```diff
@@ -204,15 +208,29 @@ def _download(self, catalog_entry, request, responder):
         :param responder:       The file-like object that nectar should write to.
         :type  responder:       Responder
         """
+        # Configure the primary downloader for alternate content sources
         importer, config = repo_controller.get_importer_by_id(catalog_entry.importer_id)
-        data = {'catalog_entry': catalog_entry, 'client_request': request}
-        download_request = nectar_request.DownloadRequest(catalog_entry.url, responder, data=data)
-        downloader = importer.get_downloader(config, catalog_entry.url,
-                                             **catalog_entry.data)
-        downloader.session = self.session
-        downloader.event_listener = StreamerListener(request, self.config)
-        downloader.download_one(download_request, events=True)
-        downloader.config.finalize()
+        primary_downloader = importer.get_downloader(config, catalog_entry.url,
+                                                     **catalog_entry.data)
+        pulp_request = request.getHeader(PULP_STREAM_REQUEST_HEADER)
+        listener = StreamerListener(request, self.config, catalog_entry, pulp_request)
+        primary_downloader.session = self.session
+        primary_downloader.event_listener = listener
+
+        # Build the alternate content source download request
+        unit_model = plugins_api.get_unit_model_by_id(catalog_entry.unit_type_id)
+        qs = unit_model.objects.filter(id=catalog_entry.unit_id).only(*unit_model.unit_key_fields)
+        unit = qs.get()
```

Comment:
.get() can raise two exceptions - are you confident that we don't need to handle those exceptions because something above us in the stack will do whatever is needed?

http://docs.mongoengine.org/en/latest/apireference.html#mongoengine.queryset.QuerySet.get